v2/tools/generator/internal/codegen/pipeline/determine_resource_ownership.go

```diff
@@ -26,32 +26,52 @@ func DetermineResourceOwnership(configuration *config.Configuration) *Stage {
 		})
 }
 
+type latestNameKey struct {
+	name  string
+	group string
+	// no version
+}
+
+func getLatestNameKey(tn astmodel.TypeName) latestNameKey {
+	return latestNameKey{
+		name:  tn.Name(),
+		group: tn.PackageReference.(astmodel.LocalPackageReference).Group(),
+	}
+}
+
+func isNewer(left, right astmodel.TypeDefinition) bool {
+	leftVersion := left.Name().PackageReference.(astmodel.LocalPackageReference).ApiVersion()
+	rightVersion := right.Name().PackageReference.(astmodel.LocalPackageReference).ApiVersion()
+
+	// versions are ASCIIbetically ordered
+	return leftVersion > rightVersion
+}
+
 func determineOwnership(definitions astmodel.TypeDefinitionSet, configuration *config.Configuration) (astmodel.TypeDefinitionSet, error) {
 	updatedDefs := make(astmodel.TypeDefinitionSet)
 
 	resources := astmodel.FindResourceDefinitions(definitions)
-	for _, def := range resources {
+	latestResources := make(map[latestNameKey]astmodel.TypeDefinition)
+	for name, resource := range resources {
+		key := getLatestNameKey(name)
+		if def, ok := latestResources[key]; ok {
+			if isNewer(resource, def) {
+				latestResources[key] = resource
+			}
+		} else {
+			latestResources[key] = resource
+		}
+	}
+
+	for _, def := range latestResources {
 		resolved, err := definitions.ResolveResourceSpecAndStatus(def)
 		if err != nil {
 			return nil, errors.Wrapf(err, "unable to find resource %s spec and status", def.Name())
 		}
 
-		childResourcePropertyTypeDef, err := extractChildResourcePropertyTypeDef(
-			definitions,
-			def.Name(),
-			resolved.SpecDef.Name(),
-			resolved.SpecType)
-		if err != nil {
-			return nil, err
-		}
-		if childResourcePropertyTypeDef == nil {
-			continue // This just means skip
-		}
+		me := def.Type().(*astmodel.ResourceType)
```

Comment:
Consider using `astmodel.AsResourcetype()` to avoid issues with wrapper meta-types.